realm/realm-library/src/main/cpp/io_realm_internal_OsRealmConfig.cpp

```diff
@@ -327,4 +327,13 @@ JNIEXPORT void JNICALL Java_io_realm_internal_OsRealmConfig_nativeSetSyncConfigS
     CATCH_STD()
 }
 
+JNIEXPORT void JNICALL Java_io_realm_internal_OsRealmConfig_nativeSetForceSyncHistory(JNIEnv*, jclass,
+                                                                                      jlong native_ptr)
+{
+    TR_ENTER_PTR(native_ptr);
+
+    auto& config = *reinterpret_cast<Realm::Config*>(native_ptr);
+    config.force_sync_history = true;
```

Comment:
```
        // FIXME: Realm Java manages sync at the Java level, so it needs to create Realms using the sync history
        //        format.
        bool force_sync_history = false;


            bool server_synchronization_mode = bool(config.sync_config) || config.force_sync_history;
            if (server_synchronization_mode) {
#if REALM_ENABLE_SYNC
                history = realm::sync::make_client_history(config.path);
#else
                REALM_TERMINATE("Realm was not built with sync enabled");
#endif
            }

```

This was added because of java was handling sync without OS at the beginning, but it seems this will be removed later. What is conclusion of supporting offline sync realm in Object Store? are we going to keep this `force_sync_history` in the future?

Reply:
AFAIK there's no plan to remove `force_sync_history` currently https://realmio.slack.com/archives/C0340J37W/p1503545777000115?thread_ts=1503491762.000111&cid=C0340J37W 
plus, [.NET](https://github.com/realm/realm-dotnet/issues/1536) may also use this option

Reply:
👍